tools/scylla-sstable.cc

```diff
@@ -2553,6 +2553,23 @@ void sstable_consumer_operation(schema_ptr schema, reader_permit permit, const s
     consumer->consume_stream_end().get();
 }
 
+void shard_of_operation(schema_ptr, reader_permit,
+                        const std::vector<sstables::shared_sstable>& sstables,
+                        sstables::sstables_manager&,
+                        const bpo::variables_map&) {
+    json_writer writer;
+    writer.StartStream();
+    for (auto& sst : sstables) {
+        writer.Key(sst->get_filename());
+        writer.StartArray();
+        for (unsigned shard_id : sst->get_shards_for_this_sstable()) {
+            writer.Uint(shard_id);
+        }
```

Comment:
Shouldn't this depend on the number of shards the source server has? I think this will work with sharder of the reconstructed schema which might be using smp=1.

Reply:
> Shouldn't this depend on the number of shards the source server has?

yes. but we don't encode the number of shards the source server has in the `Scylla` component, i think. if it's not there, could you please shed some lights on where can i find it?

> I think this will work with sharder of the reconstructed schema which might be using smp=1.

by default, the reconstructed schema uses the `smp::count` for creating its sharder, `--smp` is 1 if not specified. probably, i should add a note in the usage document to explain that user is supposed to specify `--smp` to get the correct output?

Reply:
We need a new `--shards` command-line parameter and make it required. Without this, the output of this command makes no sense.
I don't want to use `--smp`, in fact in the future I want to hide all but a few hand-selected seastar parameter when using the tools.

Reply:
ack. will add this option tomorrow.